chain/consensus/mir/state_manager.go

```diff
@@ -314,6 +319,19 @@ func (sm *StateManager) ApplyTXs(txs []*requestpb.Request) error {
 	log.With("validator", sm.id).With("epoch", sm.currentEpoch).
 		With("height", sm.height).Infof("try to create a block: msgs - %d", len(msgs))
 
+	// if new reconfiguration message, include config
+	// message in block to update on-chain membership
+	if valSet != nil {
+		// FIXME: We should pick up the genesis address and not use the default one
+		// once we move into the user-defined gateway territory
+		reconfigMsg, err := membership.NewSetMembershipMsg(genesis.DefaultIPCGatewayAddr, valSet)
+		if err != nil {
+			return err
+		}
+		// Make it a message with an empty signature, these config messages are executed implicitly
+		msgs = append(msgs, reconfigMsg)
```

Comment:
Is adding a message into msgs from the air a good idea? Would it not be better to send such a message as we do for config messages? 
Or we could add a message only if there are enough votes for it.

Reply:
Wait, I think I made a mistake, this should only be done when the vote a reconfiguration is completed, but this is not the case here, right? You may have config messages before there's consensus for the reconfiguration, right? We should move this somewhere else then :(